pkg/proxy/server.go

```diff
@@ -130,18 +132,21 @@ type Server interface {
 
 // ServerConfig defines proxy server configuration.
 type ServerConfig struct {
-	Logger        *zap.Logger
-	From          url.URL
-	To            url.URL
-	TLSInfo       transport.TLSInfo
-	DialTimeout   time.Duration
-	BufferSize    int
-	RetryInterval time.Duration
+	Logger         *zap.Logger
+	From           url.URL
+	To             url.URL
+	TLSInfo        transport.TLSInfo
+	DialTimeout    time.Duration
+	BufferSize     int
+	RetryInterval  time.Duration
+	IsForwardProxy bool
```

Comment:
Why do we need to distinguish forward proxy? 

Isn't forward proxy doing exactly the same? Maybe I missed something, but my understanding of the proposed solution was to just add additional hop.

Reply:
The original problem that the traffic is leaking is because the connection initiated from a peer will go through the destination peer's proxy then to the destination peer, and these will not be blackholed since we only drop traffic on the sender peer's proxy. 

Thus, this PR intends to fix this by adding a forward proxy on each peer. By doing so, we can perform traffic dropping just on the sender peer.

Reply:
Ok, I thought about this. As you mentioned at the beginning one proxy functions as reverse-proxy other as forward proxy. They use somewhat different code.

Reply:
I will put up another final solution to reduce the total proxy count to only 1, and see if that will work

Reply:
@serathius this PR https://github.com/etcd-io/etcd/pull/17950 is what @fuweid and I discussed offline - trying to use just 1 HTTP proxy to fulfill the blackhole requirement.